### ISSUE TITLE
[iA Writer] Feature request: Search results to show markdown preview ala Bear

### ISSUE THREAD

**Issue by gacharles23** [raycast/extensions#12968]
### Extension

https://www.raycast.com/xvvhang/raycast-ia-writer

### Description

Thank you for building this extension! 

Is there any possibility that the search feature could be enhanced to function a bit more like the Bear extension search?

- Search through notes (by title or content) and show a list of matching notes on the left and the markdown (Preview) rendered on the right. Bear examples below.

If so, that would be amazing! Thank you for the consideration.

<img width="792" alt="raycast-bear-1" src="https://github.com/raycast/extensions/assets/123527324/298df208-e387-443c-9a6d-1d39cd2039a3">

<img width="766" alt="raycast-bear-2" src="https://github.com/raycast/extensions/assets/123527324/2a4cdc8a-b124-4d29-9547-41928a93b735">

<img width="772" alt="raycast-bear-3" src="https://github.com/raycast/extensions/assets/123527324/b6fdd59f-8567-44fc-911f-3abacaea86ae">

**Comment by xvvhang** [comment]
@gacharles23 Thanks for your feedback! The extension uses iA Writer's URL Schema to achieve search feature which is quite different from Bear's API (Bear uses a sqlite to store data so it's more functional). But I've have a look at iA Writer's documents and what we can do to make the extension more functional.😀

**Comment by xvvhang** [comment]
@gacharles23 iA Writer provides only two data commands: read or write a single file. There's no way to make it happen to search the library with URL commands. I'll find out if we could make it with other solutions like a light weight database or something else.
@raycastbot keep this issue open

**Comment by gacharles23** [comment]
Yeah, that's unfortunate. I suppose I was thinking that the function could work more like the Raycast inbuilt Search Files command. That  command searches files by Name and Contents (or just Name) across all files and folders. And it displays a list of files on the left, and on the right a Preview section and a Metadata section.

The issues with the Search Files function are:
1. You can't restrict it to a directory (e.g. the iA Writer iCloud path)
2. It lists *all* filetypes, not just markdown
3. The preview displays raw text, not nicely formatted markdown

So rather than trying to build something iA Writer specific that would be severely limited by the two URL commands offered, perhaps a more generic version can be built that simply reads markdown files in a given directory -- an enhanced version of Search Files.

I suppose such a plugin would probably be useful not just for iA Writer, but any app dealing with disk-based markdown notes (e.g. Panda, Obsidian, Typora, iA Writer, ...). Although the actions would likely need to be app specific (e.g. Open file in xyz). Hmm ... maybe not. Perhaps just an Open command, and the user can pre-decide which app opens `.md` files.

In any case, thank you for giving this consideration.

**Comment by xvvhang** [comment]
@gacharles23 Great idea. It would be the best if we could configure Raycast's search file command (filter file types and preview files). Maybe we should give some feedback to Raycast.
